drivers/bridge/bridge.go

```diff
@@ -68,9 +66,13 @@ type driver struct {
 	sync.Mutex
 }
 
-func init() {
+func initializeComponents(config *Configuration) {
 	ipAllocator = ipallocator.New()
-	initPortMapper()
+	if pa := config.PortAllocator; pa != nil {
+		portMapper = portmapper.NewWithPortAllocator(pa)
```

Comment:
If we agree on making it a singleton, then most of the changes in this patch will be changed anyways.